Api Split

Vampire opened the pull request:
- Make ChannelCategoryBuilder an instantiatable class instead of an interface
- Make ServerTextChannelBuilder an instantiatable class instead of an interface
- Make ServerVoiceChannelBuilder an instantiatable class instead of an interface
- Make CustomEmojiBuilder an instantiatable class instead of an interface
- Make WebhookBuilder an instantiatable class instead of an interface
- Make ServerBuilder an instantiatable class instead of an interface
- Make RoleBuilder implementation-independent
- Make InviteBuilder implementation-independent

Vampire commented:
Added
- Fix exception validation
- Make AccountUpdater an instantiatable class instead of an interface
- Make GroupChannelUpdater an instantiatable class instead of an interface
- Make ServerChannelUpdater, ServerTextChannelUpdater and ServerVoiceChannelUpdater instantiatable classes instead of interfaces
- Make CustomEmojiUpdater an instantiatable class instead of an interface
- Make RoleUpdater an instantiatable class instead of an interface
- Make ServerUpdater an instantiatable class instead of an interface
- Make WebhookUpdater an instantiatable class instead of an interface

Vampire commented:
Added
- Rename DiscordApiFactory to DiscordApiBuilderDelegate
- Rename EmbedFactory to EmbedBuilderDelegate
- Rename MessageFactory to MessageBuilderDelegate
- Rename PermissionsFactory to PermissionsBuilderDelegate
- Rename FactoryBuilder to DelegateFactory
- Fix exception validation
- Make AccountUpdater an instantiatable class instead of an interface
- Make GroupChannelUpdater an instantiatable class instead of an interface
- Make ServerChannelUpdater, ServerTextChannelUpdater and ServerVoiceChannelUpdater instantiatable classes instead of interfaces
- Make CustomEmojiUpdater an instantiatable class instead of an interface
- Make RoleUpdater an instantiatable class instead of an interface
- Make ServerUpdater an instantiatable class instead of an interface
- Make WebhookUpdater an instantiatable class instead of an interface

BtoBastian commented:
`ImplServerTextChannelUpdater.java.bak`?

BtoBastian commented:
Maybe we should make clear in the JavaDocs that the delegates are not intended to be used by the user?
```java
/**
 * This interfaces is used by the {@link XyzBuilder} class internally.
 * You usually don't want to interact with this interface.
 */
public interface XyzBuilderDelegate { /* ... */ }
```
Something like this (I'm obviously not very good at writing JavaDoc comments ^^).

Vampire commented on src/main/java/org/javacord/entity/channel/ServerChannelUpdater.java:
```diff
@@ -3,39 +3,70 @@
-     *                    If you want to update the position based on other channels, make sure to use
-     *                    {@link ServerChannel#getRawPosition()} instead of {@link ServerChannel#getPosition()}!
+     * If you want to update the position based on other channels, make sure to use
+     * {@link ServerChannel#getRawPosition()} instead of {@link ServerChannel#getPosition()}!
```
Nope, IJ doing it, reverted.

Vampire commented:
`ImplServerTextChannelUpdater.java.bak`: totally unintentional, removed
JavaDoc comments: I now updated all to be consistent with how you did it for `EmbedBuilderDelegate`.

Vampire commented:
Added
- Enable the fallback logger to show trace logging
- Make LoggerUtil implementation-independent
- Make DiscordApi implementation-independent
- Make Message implementation-independent
